securedrop_export/export.py

```diff
@@ -0,0 +1,331 @@
+        except Exception as e:
+            print (e)
+            msg = "Error opening export bundle: "
+            self.exit_gracefully(msg, e=e)
+
+
+    def unlock_luks_volume(self, encryption_key):
+        # the luks device is not already unlocked
+        if not os.path.exists(os.path.join("/dev/mapper/", self.encrypted_device)):
+            p = subprocess.Popen(
+                ["sudo", "cryptsetup", "luksOpen", self.device, self.encrypted_device],
+                stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            p.communicate(input=str.encode(encryption_key, "utf-8"))
+            rc = p.returncode
+            if rc != 0:
+                msg = "Bad passphrase or luks error."
+                self.exit_gracefully(msg)
+
+
+    def mount_volume(self):
+        # mount target not created
+        if not os.path.exists(self.mountpoint):
+            subprocess.check_call(["sudo", "mkdir", self.mountpoint])
+        try:
+            subprocess.check_call(
+                [
+                    "sudo",
+                    "mount",
+                    os.path.join("/dev/mapper/", self.encrypted_device),
+                    self.mountpoint,
+                ]
+            )
+            subprocess.check_call(
+                [
+                    "sudo", 
+                    "chown", 
+                    "-R", "user:user", self.mountpoint,
+                ]
+            )
+        except subprocess.CalledProcessError as e:
+            # clean up
+            subprocess.check_call(["sudo", "cryptsetup", "luksClose", self.encrypted_device])
+            subprocess.check_call(["rm", "-rf", self.tmpdir]) # TODO - why not srm?
+            msg = "An error occurred while mounting disk: "
+            self.exit_gracefully(msg, e=e)
+
+
+    def copy_submission(self):
+        # move files to drive (overwrites files with same filename) and unmount drive
+        try:
+            target_path = os.path.join(self.mountpoint, self.target_dirname)
+            subprocess.check_call(["mkdir", target_path])
+            export_data = os.path.join(
+                self.tmpdir, self.submission_dirname, "export_data/"
+            )
+            subprocess.check_call(["cp", "-r", export_data, target_path])
+            popup_message("Files exported successfully to disk.")
+        except (subprocess.CalledProcessError, OSError) as e:
+            msg = "Error writing to disk:"
+            self.exit_gracefully(msg, e=e)
+        finally:
+            # Finally, we sync the filesystem, unmount the drive and lock the
+            # luks volume, and exit 0
+            subprocess.check_call(["sync"])
+            subprocess.check_call(["sudo", "umount", self.mountpoint])
+            subprocess.check_call(["sudo", "cryptsetup", "luksClose", self.encrypted_device])
+            subprocess.check_call(["rm", "-rf", self.tmpdir])
```

Comment:
Also maybe redundant per https://github.com/freedomofpress/securedrop-export/pull/1/files#diff-e41a261ab839c72ca53ae32069e0670fR18

Reply:
Exits immediately after, so not sure that this one is redundant. IMO It should stay pending another pass through the module to refine it a bit more.